cmd/mt-whisper-importer-writer/main.go

```diff
@@ -0,0 +1,283 @@
+	"github.com/raintank/metrictank/idx/cassandra"
+	"github.com/raintank/metrictank/mdata"
+	"github.com/raintank/metrictank/mdata/chunk"
+	"github.com/raintank/metrictank/mdata/chunk/archive"
+)
+
+var (
+	exitOnError = flag.Bool(
+		"exit-on-error",
+		true,
+		"Exit with a message when there's an error",
+	)
+	verbose = flag.Bool(
+		"verbose",
+		false,
+		"Write logs to terminal",
+	)
+	fakeAvgAggregates = flag.Bool(
+		"fake-avg-aggregates",
+		true,
+		"Generate sum/cnt series out of avg series to accomodate metrictank",
+	)
+	httpEndpoint = flag.String(
+		"http-endpoint",
+		"127.0.0.1:8080",
+		"The http endpoint to listen on",
+	)
+	cassandraAddrs = flag.String(
+		"cassandra-addrs",
+		"localhost",
+		"cassandra host (may be given multiple times as comma-separated list)",
+	)
+	cassandraKeyspace = flag.String(
+		"cassandra-keyspace",
+		"metrictank",
+		"cassandra keyspace to use for storing the metric data table",
+	)
+	cassAddr = flag.String(
+		"cass-addr",
+		"localhost",
+		"Address of cassandra host.",
+	)
+	ttlsStr = flag.String(
+		"ttls",
+		"35d",
+		"list of ttl strings used by MT separated by ':'",
+	)
+	windowFactor = flag.Int(
+		"window-factor",
+		20,
+		"the window factor be used when creating the metric table schema",
+	)
+	partitionScheme = flag.String(
+		"partition-scheme",
+		"bySeries",
+		"method used for partitioning metrics. This should match the settings of tsdb-gw. (byOrg|bySeries)",
+	)
+	uriPath = flag.String(
+		"uri-path",
+		"/chunks",
+		"the URI on which we expect chunks to get posted",
+	)
+	numPartitions = flag.Int(
+		"num-partitions",
+		1,
+		"Number of Partitions",
+	)
+	GitHash   = "(none)"
+	printLock sync.Mutex
+)
+
+type Server struct {
+	Cluster     *gocql.ClusterConfig
+	Session     *gocql.Session
+	TTLTables   mdata.TTLTables
+	Partitioner partitioner.Partitioner
+	Index       idx.MetricIndex
+}
+
+func main() {
+	cassandra.ConfigSetup()
+	flag.Parse()
+
+	cassCluster := gocql.NewCluster(*cassAddr)
+	cassCluster.Consistency = gocql.ParseConsistency("one")
+	cassCluster.Timeout = time.Second
+	cassCluster.NumConns = 2
+	cassCluster.ProtoVersion = 4
+	cassCluster.Keyspace = *cassandraKeyspace
+
+	session, err := cassCluster.CreateSession()
+	if err != nil {
+		throwError(fmt.Sprintf("Failed to create cassandra session: %s", err))
+	}
+
+	splits := strings.Split(*ttlsStr, ":")
+	ttls := make([]uint32, 0)
+	for _, split := range splits {
+		ttls = append(ttls, dur.MustParseUNsec("ttl", split))
+	}
+	ttlTables := mdata.GetTTLTables(ttls, *windowFactor, mdata.Table_name_format)
+
+	p, err := partitioner.NewKafka(*partitionScheme)
+	if err != nil {
+		throwError(fmt.Sprintf("Failed to instantiate partitioner: %s", err))
+	}
+
+	server := &Server{
+		Cluster:     cassCluster,
+		Session:     session,
+		TTLTables:   ttlTables,
+		Partitioner: p,
+		Index:       cassandra.New(),
+	}
+	cluster.Init("mt-whisper-importer-writer", GitHash, time.Now(), "http", int(80))
```

Comment:
Because if the cluster is not initialized then `clustering.Manager` will be a null pointer, so then the initialization of the `CasIdx` will break here: https://github.com/raintank/metrictank/blob/master/idx/cassandra/cassandra.go#L275